orchestrator/db/models.py

```diff
@@ -81,6 +83,20 @@ def process_result_value(self, value: datetime | None, dialect: Dialect) -> date
         return value.astimezone(timezone.utc) if value else value
 
 
+class InputStateTable(BaseModel):
+    __tablename__ = "input_state"
+
+    class InputType(enum.Enum):
+        user_input = "user_input"
+        initial_state = "initial_state"
+
+    input_state_id = mapped_column(UUIDType, primary_key=True, server_default=text("uuid_generate_v4()"), index=True)
+    process_id = mapped_column("pid", UUIDType, ForeignKey("processes.pid"), nullable=False)
+    input_state = mapped_column(pg.JSONB(), nullable=False)  # type: ignore
+    input_time = mapped_column(UtcTimestamp, server_default=text("current_timestamp()"), nullable=False)
+    input_type = mapped_column(Enum(InputType), nullable=False)
```

Comment:
I think we haven't used enums in the db models so far for a reason. However I have forgotten that reason..

Reply:
having two places to maintain the enum, always requiring a migration to add new types.
I like to keep enums only inside the codebase

Reply:
When I check the SA docs and google in general I don't see any mentions of it being a "bad" thing, so let's just try it out. If it ends up being bad we can change it back to a VARCHAR